### PR TITLE
[AMSDK-7500] Updated the doc to let the developers know that the SDK …

### PR DESCRIPTION
**Pull request by samthiru**
…will  lowercase string  like extension name, event type and event source strings.

### CODE REVIEW

File: resources/building-mobile-extensions/building-an-extension.md [hunk 1]
```diff
@@ -31,8 +31,12 @@ To create a simple extension, complete the following procedures in the order in
 
 ### A. **Create an extension class**
 
+
 The `ACPExtension`\(iOS\) or `Extension` \(Android\) class is the base class from which extensions must be derived. The `init` method \(iOS\) or the base `constructor` \(Android\) of your extension class is where you can extend the Adobe Experience Platform SDKs functionality by registering event listeners or by setting a default shared state that other modules can access.
 
+The Adobe Experience Cloud Platform SDK  will  lowercase the strings  like extension name, event type and event source strings, and  it is recommended  to compare the extension name, event type and event source strings using ignore case whenever compared.
```
Comment (rekrishn) [inline resources/building-mobile-extensions/building-an-extension.md]: Please change the sentence so that it reads:

"In the Adobe Experience Cloud Platform SDK, strings like the extension name, the event type, and the event source strings are in lower case. You should compare these strings by using ignore case."

- What is "ignore case"? If this an API, please use the code syntax.

- Looks like there are gaps between words that need to be removed.

File: resources/building-mobile-extensions/event-processing.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ An event is composed of the following information:
 * Event data, which is the structured data that is associated with this particular event.   This data is retrievable as a JSON blob.
 * Versioning information, which is used to synchronize a group of events with the associated shared data, is not directly exposed to extensions.  This information is implicitly used when getting shared states.
 
-**Tip**: Extensions use the `ACPExtensionEvent` \(iOS\) / `Event` \(Android\) class to create events and access the information the modules hold.
+**Tip**: Extensions use the `ACPExtensionEvent` \(iOS\) / `Event` \(Android\) class to create events and access the information the modules hold. The Adobe Experience Cloud Platform SDK  will  lowercase the strings  like extension name, event type and source strings, and  it is recommended  to compare the extension name, event type and event source strings using ignore case whenever compared.
```
Comment (rekrishn) [inline resources/building-mobile-extensions/event-processing.md]: - What does "access the information the modules hold" mean? (It's not clear.)
- Same change as above.

File: resources/building-mobile-extensions/building-an-extension.md [hunk 3]
```diff
@@ -31,8 +31,12 @@ To create a simple extension, complete the following procedures in the order in
 
 ### A. **Create an extension class**
 
+
 The `ACPExtension`\(iOS\) or `Extension` \(Android\) class is the base class from which extensions must be derived. The `init` method \(iOS\) or the base `constructor` \(Android\) of your extension class is where you can extend the Adobe Experience Platform SDKs functionality by registering event listeners or by setting a default shared state that other modules can access.
 
+In the Adobe Experience Cloud Platform SDK, the strings  like extension name, event type and source strings are converted and stored in lowercase. When you compare strings use java.lang.String.equalsIgnoreCase() method.
```
Comment (emdobrin) [inline resources/building-mobile-extensions/building-an-extension.md]: Let's not add this information here, but add it in the getName API description below. For example:
{% hint style="warning" %}
The name that you use to register **cannot** conflict with other registered extensions or Adobe internal modules. The extension name is considered case insensitive by the Mobile SDK.

File: resources/building-mobile-extensions/event-listeners.md [hunk 4]
```diff
@@ -1,7 +1,10 @@
 # Listening for events
 
+
 A common use case is to add an event listener to get notifications for events. The main location to add an event listener is in the `init` method, although you can add listeners by using other callbacks later. You can add the logic that you want executed when an event occurs, and for which you have a listener, in the `hear` method of your listener class.
 
+In the Adobe Experience Cloud Platform SDK, the strings  like extension name, event type and source strings are converted and stored in lowercase. When you compare strings use java.lang.String.equalsIgnoreCase() method.
```
Comment (emdobrin) [inline resources/building-mobile-extensions/event-listeners.md]: Let's move this after the next paragraph and reword it to something like:
`The event type and event source associated with the received event may be lower-cased as the Mobile SDK compares them as case insensitive values. It is recommended to use ignore case string comparison when checking the event type and event source of the event.`

File: resources/building-mobile-extensions/event-processing.md [hunk 5]
```diff
@@ -8,7 +8,9 @@ An event is composed of the following information:
 * Event data, which is the structured data that is associated with this particular event.   This data is retrievable as a JSON blob.
 * Versioning information, which is used to synchronize a group of events with the associated shared data, is not directly exposed to extensions.  This information is implicitly used when getting shared states.
 
-**Tip**: Extensions use the `ACPExtensionEvent` \(iOS\) / `Event` \(Android\) class to create events and access the information the modules hold.
+**Tip**: Extensions use the ACPExtensionEvent (iOS) / Event (Android) class to create events to dispatch to the extensions in order to make requests or retrieve data.
```
Comment (emdobrin) [inline resources/building-mobile-extensions/event-processing.md]: Extensions use the `ACPExtensionEvent` (iOS) / `Event` (Android) class to create events and dispatch them to the Mobile SDK event hub in order to trigger the extension business logic or to retrieve data.

File: resources/building-mobile-extensions/event-processing.md [hunk 6]
```diff
@@ -8,7 +8,9 @@ An event is composed of the following information:
...
+
+In the Adobe Experience Cloud Platform SDK, the strings  like extension name, event type and source strings are converted and stored in lowercase. When you compare strings use java.lang.String.equalsIgnoreCase() method.
```
Comment (emdobrin) [inline resources/building-mobile-extensions/event-processing.md]: I would use something similar with the above comment:
`The event type and event source may be lower-cased by the Mobile SDK as it compares them as case insensitive values. It is recommended to use ignore case string comparison when checking the event type and event source of the event.`

File: resources/building-mobile-extensions/event-listeners.md (hunk 4 above)
Comment (rekrishn) [reply resources/building-mobile-extensions/event-listeners.md]: @emdobrin and @samthiru, there is no term such as "lower cased", and the 2nd sentence uses passive text. I'd like to reword it to:

"The event type and event source that are associated with the received event can be in lower case because the Mobile SDK compares the type and the source as case insensitive values. When you check the event type and event source of the event, we recommend that you use the ignore case string comparison."

File: resources/building-mobile-extensions/event-processing.md (hunk 6 above)
Comment (rekrishn) [reply resources/building-mobile-extensions/event-processing.md]: Please use the reworded text I added above:

"The event type and event source that are associated with the received event can be in lower case because the Mobile SDK compares the type and the source as case insensitive values. When you check the event type and event source of the event, we recommend that you use the ignore case string comparison."

File: resources/building-mobile-extensions/event-processing.md (hunk 5 above)
Comment (rekrishn) [reply resources/building-mobile-extensions/event-processing.md]: Please remove "in order" and reword as follows:

"Extensions use the ACPExtensionEvent (iOS) / Event (Android) class to create events and dispatch them to the Mobile SDK event hub to trigger the extension business logic or to retrieve data."

File: resources/building-mobile-extensions/event-processing.md (hunk 6 above)
Comment (emdobrin) [reply resources/building-mobile-extensions/event-processing.md]: Thank you, @rekrishn 💯

File: resources/building-mobile-extensions/building-an-extension.md [hunk 7]
```diff
@@ -4,50 +4,56 @@
...
 
   Extensions extend the behavior of these SDKs.
 
 * Ensure that you can accomplish your goals only by using an extension.
 
   To determine your goals, think about the following questions:
 
-  * Do you need access to data that is not already exposed through the Experience Platform SDKs?    
-  * Do you need to be notified when messages will be sent, or data is being collected by the  Experience Platform SDKs?   
+  * Do you need access to data that is not already exposed via the Adobe Experience Cloud Platform SDKs?
```
Comment (emdobrin) [inline resources/building-mobile-extensions/building-an-extension.md]: Adobe Experience Platform SDKs - same below

File: resources/building-mobile-extensions/building-an-extension.md [hunk 8]
```diff
@@ -83,39 +91,38 @@ The `ACPExtension` class has the following method that you must override:
...
+@interface MyExtension : ACPExtension
+- (nullable NSString*) name;
+- (nullable NSString*) version;
+- (void) unexpectedError: (nonnull NSError*) error;
+- (void) onUnregister;
+@end
+```
 
-3. Provide at least an implementation for the `init` and `name` methods.
+1. Provide at least an implementation for the `init` and `name` methods.
```
Comment (emdobrin) [inline resources/building-mobile-extensions/building-an-extension.md]: Is there a reason for changing the step numbers here?

File: resources/building-mobile-extensions/building-an-extension.md (hunk 7 above)
Comment (rekrishn) [reply resources/building-mobile-extensions/building-an-extension.md]: @emdobrin, we only have to say "Adobe Experience Platform SDKs" once at the beginning of the documentation. After that, all instances should be "Experience Platform SDKs."
